### PR TITLE
RvR: Set up metadata/password/dhcp server on gateway IP instead of guest IP in RVR

### PR DESCRIPTION
**Pull request by ustcweizhou**
## Description

When we create a vm in the network with redundant VRs, the lease file in the vm (for example /var/lib/dhcp/dhclient.eth0.leases) shows the dhcp-server-identifier is the guest ip (not vip/gateway) of master VR. That's the ip ipaddress where the vm fetch password and metadata from.
if we stop the master VR (then backup will be master) or restart the network with cleanup (VRs will be created), the guest ip of master VR changes so vm are not able to get metadata/ssh-key using the ips in dhcp lease file.

Setting up metadata/password/dhcp server on gateway instead of guest IP in redundant VRs will fix the issues.

FIxes: #3409 

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by ustcweizhou** [comment]
@rhtyd Here is the PR for issue #3409 
I will not be able to respond to your comments in time because I will be on holiday in the coming weeks.

**Comment by rhtyd** [comment]
Thanks @ustcweizhou I'll help review and test, if you're unavailable may extend and address any review comments myself.

File: systemvm/debian/etc/apache2/vhost.template [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<VirtualHost 10.1.1.1:80>
+<VirtualHost 10.1.1.1:8180>
```
Comment (rhtyd) [inline systemvm/debian/etc/apache2/vhost.template]: Why change the port?

File: systemvm/debian/etc/apache2/vhost.template [hunk 2]
```diff
@@ -42,7 +42,7 @@
 </VirtualHost>
 
 <IfModule mod_ssl.c>
-<VirtualHost 10.1.1.1:443>
+<VirtualHost 10.1.1.1:8443>
```
Comment (rhtyd) [inline systemvm/debian/etc/apache2/vhost.template]: Same as above - any reason to change the port or did it come from your internal branch?

**Comment by rhtyd** [comment]
@blueorangutan package

File: systemvm/debian/etc/apache2/vhost.template (hunk 1 above)
Comment (ustcweizhou) [reply systemvm/debian/etc/apache2/vhost.template]: these lines will be replaced with gateway IP and guest IP in CsApp.py
If we use :80 and :443 in this template, the first time it is ok, after that the configuration file gets mess up
if we use :8180 and :8443, the configuration file will be changed only once.

**Comment by rhtyd** [comment]
@blueorangutan test

File: systemvm/debian/etc/apache2/vhost.template (hunk 1 above)
Comment (rhtyd) [reply systemvm/debian/etc/apache2/vhost.template]: @ustcweizhou Thanks for explaining. I'll test it.

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
Looks like on 4.11 branch there is some issue, I'll kick tests against master.
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
There are some failing tests, not sure if they are related to this PR
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@ustcweizhou can you fix pylint errors? Pl see Travis job 1

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
```
./debian/opt/cloud/bin/cs/CsDhcp.py:80:27: E201 whitespace after '['
../debian/opt/cloud/bin/cs/CsDhcp.py:80:39: E202 whitespace before ']'
../debian/opt/cloud/bin/cs/CsDhcp.py:123:47: E703 statement ends with a semicolon
+'[' 1 -gt 0 ']'
+echo 'Pylint failed, please check your code'
```

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by ustcweizhou** [comment]
@rhtyd 
I removed the last commit 0ae8ef0 from this PR, which is separated from other changes.

**Comment by rhtyd** [comment]
@ustcweizhou okay I'll test one more time
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
There are large number of failures can you check @ustcweizhou?

**Comment by rhtyd** [comment]
Let me kick test again nevertheless as I noticed some errors are possibly env caused.

@blueorangutan test

**Comment by rhtyd** [comment]
This is a known issue and it's likely that we are unable to fix issue and the PR seems to be adding regressions. I've moved it to 4.13.1.0 milestone for now, if the regression/issues can be fixed we can consider including the PR towards 4.13.0.0.

**Comment by ustcweizhou** [comment]
@rhtyd can you kick off a test ?
I have rebased with 4.13, and run some integration tests, it seems good.

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test matrix

**Comment by ustcweizhou** [comment]
@rhtyd I will look into the failures.
it seems ssvm is broken by this PR.

**Comment by ustcweizhou** [comment]
@rhtyd 
I have pushed a new commit to fix ssvm.
could you please kick off another test ? thanks.

**Comment by DennisKonrad** [comment]
Hi @ustcweizhou,

do you think this will solve https://github.com/apache/cloudstack/issues/3179 ?
We have this problem with KVM+OvS with redundant VPC Offering. On current master.

I suspect the changes in this PR will at least change the behaviour of VPC also.

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test matrix

**Comment by ustcweizhou** [comment]
@DennisKonrad unfortunately I do not think this pr is helpful on fixing the issue you mentioned.

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test matrix

**Comment by andrijapanicsb** [comment]
@DaanHoogland all 3 envs failed, I wiped them to regain some resources on Trillian

**Comment by DaanHoogland** [comment]
@blueorangutan test matrix

**Comment by DaanHoogland** [comment]
build failures again "TASK [Remove previous SSH key from Project if it exists] ". Not sure if this is related to the PR, but looks like it.

**Comment by rhtyd** [comment]
Trillian test failed, needs re-run @DaanHoogland cc @andrijapanicsb 
@blueorangutan package

**Comment by andrijapanicsb** [comment]
@blueorangutan test matrix

**Comment by DaanHoogland** [comment]
@rhtyd @andrijapanicsb differnet errors at different environments above. as this had been reviewed and tested before, do we spend more time investigating?

**Comment by DaanHoogland** [comment]
ping @rhtyd @andrijapanicsb ??

**Pull request merged**